mollie.php

```diff
@@ -1203,6 +1220,128 @@ public function getTabs()
+                            'utm_nooverride' => 1,
+                            'rand' => time(),
+                            'key' => $params["customer"]->secure_key,
+                            'customerId' => $params["customer"]->id
+                        ],
+                        true
+                    )
+                ),
+                'description' => $params["order"]->reference,
+                'metadata' => [
+                    'cart_id' => $params["order"]->reference,
+                    'order_reference' => $params["order"]->reference,
+                    'secure_key' => $params["order"]->secure_key
+                ],
+            ];
+
+            if (!Mollie\Utility\EnvironmentUtility::isLocalEnvironment()) {
+                $paymentData['webhookUrl'] = $this->context->link->getModuleLink(
+                    'mollie',
+                    'webhook',
+                    [],
+                    true
+                );
+            }
+
+            $newPayment = $this->api->payments->create($paymentData);
+
+            Db::getInstance()->insert(
+                'mollie_payments',
+                [
+                    'cart_id'        => (int) $params["cart"]->id,
+                    'method'         => pSQL($params["order"]->payment),
+                    'transaction_id' => $newPayment->id,
+                    'bank_status'    => _PhpScoper5eddef0da618a\Mollie\Api\Types\PaymentStatus::STATUS_OPEN,
+                    'order_id'       => (int) $params["order"]->id,
+                    'order_reference'=> (int) $params["order"]->reference,
+                    'created_at'     => array('type' => 'sql', 'value' => 'NOW()'),
+                ]
+            );
+        }
+        return $params;
+    }
+
+    /**
+     * @param $idOrder
+     * @return string
+     * @throws Exception
+     */
+    public static function resendOrderPaymentLink($orderId)
+    {
+        $module = Module::getInstanceByName('mollie');
+        /** @var \Mollie\Repository\PaymentMethodRepository $molliePaymentRepo */
+        $molliePaymentRepo = $module->getContainer(\Mollie\Repository\PaymentMethodRepository::class);
+        $molPayment = $molliePaymentRepo->getPaymentBy('order_id', $orderId);
+        if ($molPayment['bank_status'] === \_PhpScoper5eddef0da618a\Mollie\Api\Types\OrderStatus::STATUS_COMPLETED ||
+            $molPayment['bank_status'] === \_PhpScoper5eddef0da618a\Mollie\Api\Types\OrderStatus::STATUS_PAID ||
+            $molPayment['bank_status'] === \_PhpScoper5eddef0da618a\Mollie\Api\Types\OrderStatus::STATUS_SHIPPING ||
+            $molPayment['bank_status'] === \_PhpScoper5eddef0da618a\Mollie\Api\Types\PaymentStatus::STATUS_AUTHORIZED ||
+            $molPayment['bank_status'] === \_PhpScoper5eddef0da618a\Mollie\Api\Types\PaymentStatus::STATUS_PAID) {
+            return false;
+        }
+
+        $mollie = Module::getInstanceByName('mollie');
```

Comment:
if we building something for rendering tpl files and we are building variables can we use a Presenter design pattern . Basically building and collecting variables that are used for template in a separate service